### ISSUE TITLE
Make it possible to interact with external data sources

### ISSUE THREAD

**Issue by nils-braun** [nils-braun/dask-sql#13]
Currently, all data frames need to be registered before using them in `dask-sql`.
However, it could also be interesting to have dataframes directly from S3 (or any other storage, such as hdfs), from the Hive metastore or by creating temporary views. In the background, one could create dask dataframes and use the normal registration process.

For this, we first need to come up with a good SQL syntax (which is supported by calcite) and/or an API for this.

**Comment by nils-braun** [comment]
It is now (after #55) possible to create new tables by reading in data, e.g. with

```sql
CREATE TABLE
        "nyc"
    WITH (
        format = 'csv',
        location = 'https://support.staffbase.com/hc/en-us/article_attachments/360009197031/username.csv',
        sep = ';'
    )
```

However, it is not clear so far if this suits all the needs (e.g. S3). Additionally, an interaction with hive would be interesting.

**Comment by raybellwaves** [comment]
https://github.com/BlazingDB/blazingsql/issues/937 may also be relevant

**Comment by nils-braun** [comment]
Just a small update: interaction with hive is implemented as a experimetal feature in 0.2.0 (#63).
And as https://github.com/dask/adlfs/pull/111 was implemented, it should also be possible to use `az://` as input format (although untested)

**Comment by raybellwaves** [comment]
> it should also be possible to use az:// as input format (although untested)

It'll be worth testing with a public repo

```
storage_options = {'account_name': 'azureopendatastorage'}
ddf = dd.read_parquet('az://nyctlc/green/puYear=2019/puMonth=*/*.parquet', storage_options=storage_options)
```

**Comment by raybellwaves** [comment]
see https://github.com/nils-braun/dask-sql/issues/84

**Comment by nils-braun** [comment]
Closing this issue now, as we have already integrations included and for the specific new types of integration, be better create a new issue (e.g. like #83)
